src/Compilers/Core/Portable/Compilation/NullableContextOptions.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+using System;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Represents the default state of nullable analysis in this compilation.
+    /// </summary>
+    [Flags]
+    public enum NullableContextOptions
```

Comment:
i'm not really understanding the difference between these two enums.

Reply:
One of these is for a specific point in the file, the other is for the project-level. The one at the project level has less valid options than in a file, so we decided to make it separate enums.

Reply:
> One of these is for a specific point in the file, the other is for the project-level. The one at the project level has less valid options than in a file,

Why does the one at project-level have less valid options than in a file?  As a user i would think "if i can specify this on a file by file basis, certainly i can specify it at the project level and have it apply to all files.  Right?

Reply:
See other comment for the full story, but when you're in a given file at a specific point we also want to convey whether or not that was an explicit override point in the file vs. just following the defaults.